.github/workflows/dependency-update.go-mod-tidy_and_make-package-specs.yaml

```diff
@@ -99,7 +100,14 @@ jobs:
             git add .
             git commit --message="${tidy_message}"
             git push
+            echo "changed=true" >> $GITHUB_ENV
           else
             echo 'No files changed!'
+            echo "changed=false" >> $GITHUB_ENV
           fi
           echo '🏁'
+      - uses: hmarr/auto-approve-action@v4
+        name: Auto-approve
+        if: steps.go-mod-tidy.outputs.changed == 'true' && github.event.pull_request.user.login == 'dependabot[bot]'
```

Comment:
According to the action documentation, the more or less simplest usage of the action would be the following

```yaml 
name: Auto approve

on: pull_request_target

jobs:
  auto-approve:
    runs-on: ubuntu-latest
    permissions:
      pull-requests: write
    if: github.actor == 'dependabot[bot]'
    steps:
      - uses: hmarr/auto-approve-action@v4
```

Comparing the simplest example above with this here, I am asking myself the following questions:

1. Why is it important to only approve in case of preceding changes? I was just asking myself if it may be simpler to always simply approve independant of potential preceding changes.
2. Why do we need to approve with our ci-user?

Reply:
Good points, initially I was thinking that it made sense to restrictively only approve changes by our CI.
Adressed in https://github.com/cloudfoundry/app-autoscaler-release/pull/2839/commits/98d954fc4df2c5cbe1ae676d2e67e18002001799.

I also removed two now obsolete workflows.

Reply:
Just out of curiosity, why do these two workflows became obsolete now? 🤓

Reply:
`.dependency-spec-updates.yaml` became obsolete with the introduction of `dependency-update.go-mod-tidy_and_make-package-specs.yaml` and never worked afterwards: https://github.com/cloudfoundry/app-autoscaler-release/actions/workflows/dependency-spec-updates.yaml - the green runs are where it didn't have to do anything because it was already taken care of by dependency-update.go-mod-tidy_and_make-package-specs.yaml`

Reply:
`dependency-updates-auto-merge.yaml` had two jobs: Approving `dependabot` PRs, which we disabled in favor of `renovate` and enabling automerge which `renovate` also takes care of itself, e.g. see https://github.com/cloudfoundry/app-autoscaler-release/pull/2841 : 
![image](https://github.com/cloudfoundry/app-autoscaler-release/assets/3974803/19ad8135-2d0a-4ff2-9561-0089b91827ee)

Reply:
thanks for cleaning this up! 🤝